modules/service/src/main/scala/lucuma/odb/service/GeneratorParamsService.scala

```diff
@@ -148,7 +147,7 @@ object GeneratorParamsService {
           m  <- observingModeServices.selectObservingMode(oms)
         } yield
           ps.groupBy(_.observationId).map { case (oid, oParams) =>
-            oid -> toGeneratorParams(oParams, m.get(oid))
+            oid -> toGeneratorParams(NonEmptyList.fromListUnsafe(oParams), m.get(oid))
```

Comment:
Do we know `oParams` is non-empty?

Reply:
Yes, it is created by `groupBy` which will not create an empty grouping.